app/models/legal_aid_application.rb

```diff
@@ -53,13 +53,12 @@ class LegalAidApplication < ApplicationRecord
 
   before_save :set_open_banking_consent_choice_at
   before_create :create_app_ref
-  after_create do
+  after_commit on: :create do
     ActiveSupport::Notifications.instrument "dashboard.application_created", id:, state:
   end
 
-  after_save do
+  after_commit do
     ActiveSupport::Notifications.instrument "dashboard.declined_open_banking" if saved_change_to_open_banking_consent?
-    ActiveSupport::Notifications.instrument("dashboard.provider_updated", provider_id: provider.id) if proc { |laa| laa.state }.eql?(:assessment_submitted)
```

Comment:
need to understand this removal better 🤔

Reply:
I speak about it in the commit message c9b53b658dfb65033f187a34708806564e8ffa45. 

As far as I could tell, the notification can never be sent, because `proc { |laa| laa.state }.eql?(:assessment_submitted)` always evaluates to `false`. The code was untested, so I deleted the code.